source/adios2/toolkit/format/dataman/DataManSerializer.tcc

```diff
@@ -257,8 +257,8 @@ int DataManSerializer::GetData(T *outputData, const std::string &varName,
                 input_data += j.position;
             }
 
-            if (j.shape.size() > 0 and j.shape[0] > 1 and j.start.size() > 0 and
-                j.start.size() == j.count.size() and
+            if (j.shape.size() > 0 and (j.shape[0] > 1 or j.shape[1] > 1) and
```

Comment:
Correct me if I'm wrong, but if j.shape.size() == 1, won't a hard-coded reference to j.shape[1] be a reference to unallocated memory?  That seems like a bad idea...

Reply:
If ```j.shape.size() == 1```  and ```j.shape[0] > 1``` the second condition ```j.shape[1] > 1``` will not be checked by a compiler. The only problem here could be single values presented as arrays like [1]

Reply:
... but it goes to memcopy branch

Reply:
And if j.shape.size() == 1 and  j.shape[0] == 1 the second condition will be checked and will result in accessing unallocated memory, right?    But I think you're missing the point.  This code was structured badly before this PR, resulting in a silent failure to move data.  That your proposed changes are predicated on checking the first two dimensions of an N-dimensional array is a pretty solid indication that this isn't a robust fix.  How can you write this so that it's actually robust?

Reply:
If there should be a wrong memory access, I am wondering why CI is not showing it?

Reply:
Unfortunately, msan is particularly computationally expensive and if we were to run all our tests under it, they would take several hours to complete. Our tier of CI times out after an hour, so we can run only a small subset of tests in the sanitizers.  Fortunately, this case is obvious from inspection, but I do sometimes compile with msan or use valgrind to verify things that don't run in CI.